supervisor/container.go

```diff
@@ -226,30 +236,41 @@ func (c *Container) addProcess(processId, stdin, stdout, stderr string, spec *sp
 	c.ownerPod.Processes[processId] = p
 	c.Processes[processId] = p
 
-	e := Event{
-		ID:        c.Id,
-		Type:      EventProcessStart,
-		Timestamp: time.Now(),
-		PID:       processId,
-	}
-	c.ownerPod.sv.Events.notifySubscribers(e)
+	// Wait for the VM to acknowledge the process has attached to avoid a
+	// libcontainer deadlock.
+	resultCh := make(chan error)
 
 	go func() {
 		c.ownerPod.podStatus.AddExec(c.Id, processId, "", spec.Terminal)
 		err := c.ownerPod.vm.AddProcess(c.Id, processId, spec.Terminal, spec.Args, spec.Env, spec.Cwd, p.stdio)
```

Comment:
I'll see if I can concoct a test to catch this situation so I can avoid it (my point testing was based on the noted ticket, which of course exits quite quickly).